Mage/src/main/java/mage/game/tournament/TournamentPlayer.java

```diff
@@ -101,25 +101,22 @@ public boolean updateDeck(Deck deck) {
     }
 
     public Deck generateDeck() {
-        // user passed to submit deck in time
-        // all all cards to deck
-        deck.getCards().addAll(deck.getSideboard());
```

Comment:
Important changes here:
- before: auto-add all sideboard cards (player can't hide cards or trim a deck size to lower number of cards)
- after: ignore sideboard cards and allow to trim a deck size

Is it fine for real games (e.g. for cheating, etc)? Cheat example: tourney with 60 cards deck -- user can trim it to 40 cards and increase card rates.

Real deck size can be found by deck validator, search by `deckMinSize = deckValidator`

Reply:
The `TournamentPlayer` class doesn't seem to have any way to access a correct `deckValidator` as is; that would probably require substantial refactoring.

I don't believe this is necessary as this method is only called for the "Constructing" phase of Limited tournaments. (A different method is used when sideboarding during a match.) There is only one deckValidator for Limited and it sets 40 as the minimum deck size. Constructed events require players to join with a premade deck where there is no way to timeout and autosubmit. Thus I think it should be okay to hardcode 40 here. If a future limited format requires a different minimum deck size, some refactoring would be necessary. I could be misunderstanding but I don't see a way for someone to cheat as a result of this change.

Reply:
Alright, using the deck validator now.